From jmm@debian.org  Wed Jan 21 16:31:28 2009
From: Moritz Muehlenhoff <jmm@debian.org>
Date: Wed, 21 Jan 2009 22:00:40 +0100
Subject: Staging: wlan-ng: Remove use of __WLAN_ATTRIB_PACK__
To: greg@kroah.com
Cc: Moritz Muehlenhoff <jmm@debian.org>
Message-ID: <1232571646-32579-1-git-send-email-jmm@debian.org>


Replace all ocurrances of the __WLAN_ATTRIB_PACK__ from wlan_compat.h
by __attribute__((packed)) and remove it afterwards.

Signed-off-by: Moritz Muehlenhoff <jmm@debian.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/wlan-ng/hfa384x.h          |  310 ++++++++++++++---------------
 drivers/staging/wlan-ng/p80211conv.h       |    6 
 drivers/staging/wlan-ng/p80211hdr.h        |    6 
 drivers/staging/wlan-ng/p80211ioctl.h      |    2 
 drivers/staging/wlan-ng/p80211metastruct.h |   30 +-
 drivers/staging/wlan-ng/p80211mgmt.h       |   18 -
 drivers/staging/wlan-ng/p80211msg.h        |    4 
 drivers/staging/wlan-ng/p80211types.h      |   34 +--
 drivers/staging/wlan-ng/wlan_compat.h      |    5 
 9 files changed, 205 insertions(+), 210 deletions(-)

--- a/drivers/staging/wlan-ng/hfa384x.h
+++ b/drivers/staging/wlan-ng/hfa384x.h
@@ -893,13 +893,13 @@ typedef struct hfa384x_bytestr
 {
 	u16	len;
 	u8	data[0];
-} __WLAN_ATTRIB_PACK__ hfa384x_bytestr_t;
+} __attribute__((packed)) hfa384x_bytestr_t;
 
 typedef struct hfa384x_bytestr32
 {
 	u16	len;
 	u8	data[32];
-} __WLAN_ATTRIB_PACK__ hfa384x_bytestr32_t;
+} __attribute__((packed)) hfa384x_bytestr32_t;
 
 /*--------------------------------------------------------------------
 Configuration Record Structures:
@@ -912,21 +912,21 @@ typedef struct hfa384x_record
 {
 	u16	reclen;
 	u16	rid;
-} __WLAN_ATTRIB_PACK__ hfa384x_rec_t;
+} __attribute__((packed)) hfa384x_rec_t;
 
 typedef struct hfa384x_record16
 {
 	u16	reclen;
 	u16	rid;
 	u16	val;
-} __WLAN_ATTRIB_PACK__ hfa384x_rec16_t;
+} __attribute__((packed)) hfa384x_rec16_t;
 
 typedef struct hfa384x_record32
 {
 	u16	reclen;
 	u16	rid;
 	u32	val;
-} __WLAN_ATTRIB_PACK__ hfa384x_rec32;
+} __attribute__((packed)) hfa384x_rec32;
 
 /*-- Hardware/Firmware Component Information ----------*/
 typedef struct hfa384x_compident
@@ -935,7 +935,7 @@ typedef struct hfa384x_compident
 	u16	variant;
 	u16	major;
 	u16	minor;
-} __WLAN_ATTRIB_PACK__ hfa384x_compident_t;
+} __attribute__((packed)) hfa384x_compident_t;
 
 typedef struct hfa384x_caplevel
 {
@@ -944,79 +944,79 @@ typedef struct hfa384x_caplevel
 	u16	variant;
 	u16	bottom;
 	u16	top;
-} __WLAN_ATTRIB_PACK__ hfa384x_caplevel_t;
+} __attribute__((packed)) hfa384x_caplevel_t;
 
 /*-- Configuration Record: cnfPortType --*/
 typedef struct hfa384x_cnfPortType
 {
 	u16	cnfPortType;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfPortType_t;
+} __attribute__((packed)) hfa384x_cnfPortType_t;
 
 /*-- Configuration Record: cnfOwnMACAddress --*/
 typedef struct hfa384x_cnfOwnMACAddress
 {
 	u8	cnfOwnMACAddress[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnMACAddress_t;
+} __attribute__((packed)) hfa384x_cnfOwnMACAddress_t;
 
 /*-- Configuration Record: cnfDesiredSSID --*/
 typedef struct hfa384x_cnfDesiredSSID
 {
 	u8	cnfDesiredSSID[34];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfDesiredSSID_t;
+} __attribute__((packed)) hfa384x_cnfDesiredSSID_t;
 
 /*-- Configuration Record: cnfOwnChannel --*/
 typedef struct hfa384x_cnfOwnChannel
 {
 	u16	cnfOwnChannel;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnChannel_t;
+} __attribute__((packed)) hfa384x_cnfOwnChannel_t;
 
 /*-- Configuration Record: cnfOwnSSID --*/
 typedef struct hfa384x_cnfOwnSSID
 {
 	u8	cnfOwnSSID[34];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnSSID_t;
+} __attribute__((packed)) hfa384x_cnfOwnSSID_t;
 
 /*-- Configuration Record: cnfOwnATIMWindow --*/
 typedef struct hfa384x_cnfOwnATIMWindow
 {
 	u16	cnfOwnATIMWindow;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnATIMWindow_t;
+} __attribute__((packed)) hfa384x_cnfOwnATIMWindow_t;
 
 /*-- Configuration Record: cnfSystemScale --*/
 typedef struct hfa384x_cnfSystemScale
 {
 	u16	cnfSystemScale;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfSystemScale_t;
+} __attribute__((packed)) hfa384x_cnfSystemScale_t;
 
 /*-- Configuration Record: cnfMaxDataLength --*/
 typedef struct hfa384x_cnfMaxDataLength
 {
 	u16	cnfMaxDataLength;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfMaxDataLength_t;
+} __attribute__((packed)) hfa384x_cnfMaxDataLength_t;
 
 /*-- Configuration Record: cnfWDSAddress --*/
 typedef struct hfa384x_cnfWDSAddress
 {
 	u8	cnfWDSAddress[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfWDSAddress_t;
+} __attribute__((packed)) hfa384x_cnfWDSAddress_t;
 
 /*-- Configuration Record: cnfPMEnabled --*/
 typedef struct hfa384x_cnfPMEnabled
 {
 	u16	cnfPMEnabled;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfPMEnabled_t;
+} __attribute__((packed)) hfa384x_cnfPMEnabled_t;
 
 /*-- Configuration Record: cnfPMEPS --*/
 typedef struct hfa384x_cnfPMEPS
 {
 	u16	cnfPMEPS;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfPMEPS_t;
+} __attribute__((packed)) hfa384x_cnfPMEPS_t;
 
 /*-- Configuration Record: cnfMulticastReceive --*/
 typedef struct hfa384x_cnfMulticastReceive
 {
 	u16	cnfMulticastReceive;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfMulticastReceive_t;
+} __attribute__((packed)) hfa384x_cnfMulticastReceive_t;
 
 /*-- Configuration Record: cnfAuthentication --*/
 #define HFA384x_CNFAUTHENTICATION_OPENSYSTEM	0x0001
@@ -1027,37 +1027,37 @@ typedef struct hfa384x_cnfMulticastRecei
 typedef struct hfa384x_cnfMaxSleepDuration
 {
 	u16	cnfMaxSleepDuration;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfMaxSleepDuration_t;
+} __attribute__((packed)) hfa384x_cnfMaxSleepDuration_t;
 
 /*-- Configuration Record: cnfPMHoldoverDuration --*/
 typedef struct hfa384x_cnfPMHoldoverDuration
 {
 	u16	cnfPMHoldoverDuration;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfPMHoldoverDuration_t;
+} __attribute__((packed)) hfa384x_cnfPMHoldoverDuration_t;
 
 /*-- Configuration Record: cnfOwnName --*/
 typedef struct hfa384x_cnfOwnName
 {
 	u8	cnfOwnName[34];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnName_t;
+} __attribute__((packed)) hfa384x_cnfOwnName_t;
 
 /*-- Configuration Record: cnfOwnDTIMPeriod --*/
 typedef struct hfa384x_cnfOwnDTIMPeriod
 {
 	u16	cnfOwnDTIMPeriod;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfOwnDTIMPeriod_t;
+} __attribute__((packed)) hfa384x_cnfOwnDTIMPeriod_t;
 
 /*-- Configuration Record: cnfWDSAddress --*/
 typedef struct hfa384x_cnfWDSAddressN
 {
 	u8	cnfWDSAddress[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfWDSAddressN_t;
+} __attribute__((packed)) hfa384x_cnfWDSAddressN_t;
 
 /*-- Configuration Record: cnfMulticastPMBuffering --*/
 typedef struct hfa384x_cnfMulticastPMBuffering
 {
 	u16	cnfMulticastPMBuffering;
-} __WLAN_ATTRIB_PACK__ hfa384x_cnfMulticastPMBuffering_t;
+} __attribute__((packed)) hfa384x_cnfMulticastPMBuffering_t;
 
 /*--------------------------------------------------------------------
 Configuration Record Structures:
@@ -1068,13 +1068,13 @@ Configuration Record Structures:
 typedef struct hfa384x_GroupAddresses
 {
 	u8	MACAddress[16][6];
-} __WLAN_ATTRIB_PACK__ hfa384x_GroupAddresses_t;
+} __attribute__((packed)) hfa384x_GroupAddresses_t;
 
 /*-- Configuration Record: CreateIBSS --*/
 typedef struct hfa384x_CreateIBSS
 {
 	u16	CreateIBSS;
-} __WLAN_ATTRIB_PACK__ hfa384x_CreateIBSS_t;
+} __attribute__((packed)) hfa384x_CreateIBSS_t;
 
 #define HFA384x_CREATEIBSS_JOINCREATEIBSS          0
 #define HFA384x_CREATEIBSS_JOINESS_JOINCREATEIBSS  1
@@ -1085,32 +1085,32 @@ typedef struct hfa384x_CreateIBSS
 typedef struct hfa384x_FragmentationThreshold
 {
 	u16	FragmentationThreshold;
-} __WLAN_ATTRIB_PACK__ hfa384x_FragmentationThreshold_t;
+} __attribute__((packed)) hfa384x_FragmentationThreshold_t;
 
 /*-- Configuration Record: RTSThreshold --*/
 typedef struct hfa384x_RTSThreshold
 {
 	u16	RTSThreshold;
-} __WLAN_ATTRIB_PACK__ hfa384x_RTSThreshold_t;
+} __attribute__((packed)) hfa384x_RTSThreshold_t;
 
 /*-- Configuration Record: TxRateControl --*/
 typedef struct hfa384x_TxRateControl
 {
 	u16	TxRateControl;
-} __WLAN_ATTRIB_PACK__ hfa384x_TxRateControl_t;
+} __attribute__((packed)) hfa384x_TxRateControl_t;
 
 /*-- Configuration Record: PromiscuousMode --*/
 typedef struct hfa384x_PromiscuousMode
 {
 	u16	PromiscuousMode;
-} __WLAN_ATTRIB_PACK__ hfa384x_PromiscuousMode_t;
+} __attribute__((packed)) hfa384x_PromiscuousMode_t;
 
 /*-- Configuration Record: ScanRequest (data portion only) --*/
 typedef struct hfa384x_ScanRequest_data
 {
 	u16	channelList;
 	u16	txRate;
-} __WLAN_ATTRIB_PACK__ hfa384x_ScanRequest_data_t;
+} __attribute__((packed)) hfa384x_ScanRequest_data_t;
 
 /*-- Configuration Record: HostScanRequest (data portion only) --*/
 typedef struct hfa384x_HostScanRequest_data
@@ -1118,14 +1118,14 @@ typedef struct hfa384x_HostScanRequest_d
 	u16	channelList;
 	u16	txRate;
 	hfa384x_bytestr32_t ssid;
-} __WLAN_ATTRIB_PACK__ hfa384x_HostScanRequest_data_t;
+} __attribute__((packed)) hfa384x_HostScanRequest_data_t;
 
 /*-- Configuration Record: JoinRequest (data portion only) --*/
 typedef struct hfa384x_JoinRequest_data
 {
 	u8	bssid[WLAN_BSSID_LEN];
 	u16	channel;
-} __WLAN_ATTRIB_PACK__ hfa384x_JoinRequest_data_t;
+} __attribute__((packed)) hfa384x_JoinRequest_data_t;
 
 /*-- Configuration Record: authenticateStation (data portion only) --*/
 typedef struct hfa384x_authenticateStation_data
@@ -1133,7 +1133,7 @@ typedef struct hfa384x_authenticateStati
 	u8	address[WLAN_ADDR_LEN];
 	u16	status;
 	u16	algorithm;
-} __WLAN_ATTRIB_PACK__ hfa384x_authenticateStation_data_t;
+} __attribute__((packed)) hfa384x_authenticateStation_data_t;
 
 /*-- Configuration Record: associateStation (data portion only) --*/
 typedef struct hfa384x_associateStation_data
@@ -1141,14 +1141,14 @@ typedef struct hfa384x_associateStation_
 	u8	address[WLAN_ADDR_LEN];
 	u16	status;
 	u16	type;
-} __WLAN_ATTRIB_PACK__ hfa384x_associateStation_data_t;
+} __attribute__((packed)) hfa384x_associateStation_data_t;
 
 /*-- Configuration Record: ChannelInfoRequest (data portion only) --*/
 typedef struct hfa384x_ChannelInfoRequest_data
 {
 	u16	channelList;
 	u16	channelDwellTime;
-} __WLAN_ATTRIB_PACK__ hfa384x_ChannelInfoRequest_data_t;
+} __attribute__((packed)) hfa384x_ChannelInfoRequest_data_t;
 
 /*-- Configuration Record: WEPKeyMapping (data portion only) --*/
 typedef struct hfa384x_WEPKeyMapping
@@ -1158,14 +1158,14 @@ typedef struct hfa384x_WEPKeyMapping
 	u8 	key[16];
 	u8 	mic_transmit_key[4];
 	u8 	mic_receive_key[4];
-} __WLAN_ATTRIB_PACK__ hfa384x_WEPKeyMapping_t;
+} __attribute__((packed)) hfa384x_WEPKeyMapping_t;
 
 /*-- Configuration Record: WPAData       (data portion only) --*/
 typedef struct hfa384x_WPAData
 {
 	u16	datalen;
         u8 	data[0]; // max 80
-} __WLAN_ATTRIB_PACK__ hfa384x_WPAData_t;
+} __attribute__((packed)) hfa384x_WPAData_t;
 
 /*--------------------------------------------------------------------
 Configuration Record Structures: Behavior Parameters
@@ -1175,7 +1175,7 @@ Configuration Record Structures: Behavio
 typedef struct hfa384x_TickTime
 {
 	u16	TickTime;
-} __WLAN_ATTRIB_PACK__ hfa384x_TickTime_t;
+} __attribute__((packed)) hfa384x_TickTime_t;
 
 /*--------------------------------------------------------------------
 Information Record Structures: NIC Information
@@ -1185,7 +1185,7 @@ Information Record Structures: NIC Infor
 typedef struct hfa384x_MaxLoadTime
 {
 	u16	MaxLoadTime;
-} __WLAN_ATTRIB_PACK__ hfa384x_MaxLoadTime_t;
+} __attribute__((packed)) hfa384x_MaxLoadTime_t;
 
 /*-- Information Record: DownLoadBuffer --*/
 /* NOTE: The page and offset are in AUX format */
@@ -1194,7 +1194,7 @@ typedef struct hfa384x_downloadbuffer
 	u16	page;
 	u16	offset;
 	u16	len;
-} __WLAN_ATTRIB_PACK__ hfa384x_downloadbuffer_t;
+} __attribute__((packed)) hfa384x_downloadbuffer_t;
 
 /*-- Information Record: PRIIdentity --*/
 typedef struct hfa384x_PRIIdentity
@@ -1203,7 +1203,7 @@ typedef struct hfa384x_PRIIdentity
 	u16	PRIVariant;
 	u16	PRIMajorVersion;
 	u16	PRIMinorVersion;
-} __WLAN_ATTRIB_PACK__ hfa384x_PRIIdentity_t;
+} __attribute__((packed)) hfa384x_PRIIdentity_t;
 
 /*-- Information Record: PRISupRange --*/
 typedef struct hfa384x_PRISupRange
@@ -1213,7 +1213,7 @@ typedef struct hfa384x_PRISupRange
 	u16	PRIVariant;
 	u16	PRIBottom;
 	u16	PRITop;
-} __WLAN_ATTRIB_PACK__ hfa384x_PRISupRange_t;
+} __attribute__((packed)) hfa384x_PRISupRange_t;
 
 /*-- Information Record: CFIActRanges --*/
 typedef struct hfa384x_CFIActRanges
@@ -1223,13 +1223,13 @@ typedef struct hfa384x_CFIActRanges
 	u16	CFIVariant;
 	u16	CFIBottom;
 	u16	CFITop;
-} __WLAN_ATTRIB_PACK__ hfa384x_CFIActRanges_t;
+} __attribute__((packed)) hfa384x_CFIActRanges_t;
 
 /*-- Information Record: NICSerialNumber --*/
 typedef struct hfa384x_NICSerialNumber
 {
 	u8	NICSerialNumber[12];
-} __WLAN_ATTRIB_PACK__ hfa384x_NICSerialNumber_t;
+} __attribute__((packed)) hfa384x_NICSerialNumber_t;
 
 /*-- Information Record: NICIdentity --*/
 typedef struct hfa384x_NICIdentity
@@ -1238,7 +1238,7 @@ typedef struct hfa384x_NICIdentity
 	u16	NICVariant;
 	u16	NICMajorVersion;
 	u16	NICMinorVersion;
-} __WLAN_ATTRIB_PACK__ hfa384x_NICIdentity_t;
+} __attribute__((packed)) hfa384x_NICIdentity_t;
 
 /*-- Information Record: MFISupRange --*/
 typedef struct hfa384x_MFISupRange
@@ -1248,7 +1248,7 @@ typedef struct hfa384x_MFISupRange
 	u16	MFIVariant;
 	u16	MFIBottom;
 	u16	MFITop;
-} __WLAN_ATTRIB_PACK__ hfa384x_MFISupRange_t;
+} __attribute__((packed)) hfa384x_MFISupRange_t;
 
 /*-- Information Record: CFISupRange --*/
 typedef struct hfa384x_CFISupRange
@@ -1258,44 +1258,44 @@ typedef struct hfa384x_CFISupRange
 	u16	CFIVariant;
 	u16	CFIBottom;
 	u16	CFITop;
-} __WLAN_ATTRIB_PACK__ hfa384x_CFISupRange_t;
+} __attribute__((packed)) hfa384x_CFISupRange_t;
 
 /*-- Information Record: BUILDSEQ:BuildSeq --*/
 typedef struct hfa384x_BuildSeq {
 	u16	primary;
 	u16	secondary;
-} __WLAN_ATTRIB_PACK__ hfa384x_BuildSeq_t;
+} __attribute__((packed)) hfa384x_BuildSeq_t;
 
 /*-- Information Record: FWID --*/
 #define HFA384x_FWID_LEN	14
 typedef struct hfa384x_FWID {
 	u8	primary[HFA384x_FWID_LEN];
 	u8	secondary[HFA384x_FWID_LEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_FWID_t;
+} __attribute__((packed)) hfa384x_FWID_t;
 
 /*-- Information Record: ChannelList --*/
 typedef struct hfa384x_ChannelList
 {
 	u16	ChannelList;
-} __WLAN_ATTRIB_PACK__ hfa384x_ChannelList_t;
+} __attribute__((packed)) hfa384x_ChannelList_t;
 
 /*-- Information Record: RegulatoryDomains --*/
 typedef struct hfa384x_RegulatoryDomains
 {
 	u8	RegulatoryDomains[12];
-} __WLAN_ATTRIB_PACK__ hfa384x_RegulatoryDomains_t;
+} __attribute__((packed)) hfa384x_RegulatoryDomains_t;
 
 /*-- Information Record: TempType --*/
 typedef struct hfa384x_TempType
 {
 	u16	TempType;
-} __WLAN_ATTRIB_PACK__ hfa384x_TempType_t;
+} __attribute__((packed)) hfa384x_TempType_t;
 
 /*-- Information Record: CIS --*/
 typedef struct hfa384x_CIS
 {
 	u8	CIS[480];
-} __WLAN_ATTRIB_PACK__ hfa384x_CIS_t;
+} __attribute__((packed)) hfa384x_CIS_t;
 
 /*-- Information Record: STAIdentity --*/
 typedef struct hfa384x_STAIdentity
@@ -1304,7 +1304,7 @@ typedef struct hfa384x_STAIdentity
 	u16	STAVariant;
 	u16	STAMajorVersion;
 	u16	STAMinorVersion;
-} __WLAN_ATTRIB_PACK__ hfa384x_STAIdentity_t;
+} __attribute__((packed)) hfa384x_STAIdentity_t;
 
 /*-- Information Record: STASupRange --*/
 typedef struct hfa384x_STASupRange
@@ -1314,7 +1314,7 @@ typedef struct hfa384x_STASupRange
 	u16	STAVariant;
 	u16	STABottom;
 	u16	STATop;
-} __WLAN_ATTRIB_PACK__ hfa384x_STASupRange_t;
+} __attribute__((packed)) hfa384x_STASupRange_t;
 
 /*-- Information Record: MFIActRanges --*/
 typedef struct hfa384x_MFIActRanges
@@ -1324,7 +1324,7 @@ typedef struct hfa384x_MFIActRanges
 	u16	MFIVariant;
 	u16	MFIBottom;
 	u16	MFITop;
-} __WLAN_ATTRIB_PACK__ hfa384x_MFIActRanges_t;
+} __attribute__((packed)) hfa384x_MFIActRanges_t;
 
 /*--------------------------------------------------------------------
 Information Record Structures: NIC Information
@@ -1334,7 +1334,7 @@ Information Record Structures: NIC Infor
 typedef struct hfa384x_PortStatus
 {
 	u16	PortStatus;
-} __WLAN_ATTRIB_PACK__ hfa384x_PortStatus_t;
+} __attribute__((packed)) hfa384x_PortStatus_t;
 
 #define HFA384x_PSTATUS_DISABLED	((u16)1)
 #define HFA384x_PSTATUS_SEARCHING	((u16)2)
@@ -1347,13 +1347,13 @@ typedef struct hfa384x_PortStatus
 typedef struct hfa384x_CurrentSSID
 {
 	u8	CurrentSSID[34];
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentSSID_t;
+} __attribute__((packed)) hfa384x_CurrentSSID_t;
 
 /*-- Information Record: CurrentBSSID --*/
 typedef struct hfa384x_CurrentBSSID
 {
 	u8	CurrentBSSID[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentBSSID_t;
+} __attribute__((packed)) hfa384x_CurrentBSSID_t;
 
 /*-- Information Record: commsquality --*/
 typedef struct hfa384x_commsquality
@@ -1361,7 +1361,7 @@ typedef struct hfa384x_commsquality
 	u16	CQ_currBSS;
 	u16	ASL_currBSS;
 	u16	ANL_currFC;
-} __WLAN_ATTRIB_PACK__ hfa384x_commsquality_t;
+} __attribute__((packed)) hfa384x_commsquality_t;
 
 /*-- Information Record: dmbcommsquality --*/
 typedef struct hfa384x_dbmcommsquality
@@ -1369,19 +1369,19 @@ typedef struct hfa384x_dbmcommsquality
 	u16	CQdbm_currBSS;
 	u16	ASLdbm_currBSS;
 	u16	ANLdbm_currFC;
-} __WLAN_ATTRIB_PACK__ hfa384x_dbmcommsquality_t;
+} __attribute__((packed)) hfa384x_dbmcommsquality_t;
 
 /*-- Information Record: CurrentTxRate --*/
 typedef struct hfa384x_CurrentTxRate
 {
 	u16	CurrentTxRate;
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentTxRate_t;
+} __attribute__((packed)) hfa384x_CurrentTxRate_t;
 
 /*-- Information Record: CurrentBeaconInterval --*/
 typedef struct hfa384x_CurrentBeaconInterval
 {
 	u16	CurrentBeaconInterval;
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentBeaconInterval_t;
+} __attribute__((packed)) hfa384x_CurrentBeaconInterval_t;
 
 /*-- Information Record: CurrentScaleThresholds --*/
 typedef struct hfa384x_CurrentScaleThresholds
@@ -1391,50 +1391,50 @@ typedef struct hfa384x_CurrentScaleThres
 	u16	DeferDetectThreshold;
 	u16	CellSearchThreshold; /* Stations only */
 	u16	DeadSpotThreshold; /* Stations only */
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentScaleThresholds_t;
+} __attribute__((packed)) hfa384x_CurrentScaleThresholds_t;
 
 /*-- Information Record: ProtocolRspTime --*/
 typedef struct hfa384x_ProtocolRspTime
 {
 	u16	ProtocolRspTime;
-} __WLAN_ATTRIB_PACK__ hfa384x_ProtocolRspTime_t;
+} __attribute__((packed)) hfa384x_ProtocolRspTime_t;
 
 /*-- Information Record: ShortRetryLimit --*/
 typedef struct hfa384x_ShortRetryLimit
 {
 	u16	ShortRetryLimit;
-} __WLAN_ATTRIB_PACK__ hfa384x_ShortRetryLimit_t;
+} __attribute__((packed)) hfa384x_ShortRetryLimit_t;
 
 /*-- Information Record: LongRetryLimit --*/
 typedef struct hfa384x_LongRetryLimit
 {
 	u16	LongRetryLimit;
-} __WLAN_ATTRIB_PACK__ hfa384x_LongRetryLimit_t;
+} __attribute__((packed)) hfa384x_LongRetryLimit_t;
 
 /*-- Information Record: MaxTransmitLifetime --*/
 typedef struct hfa384x_MaxTransmitLifetime
 {
 	u16	MaxTransmitLifetime;
-} __WLAN_ATTRIB_PACK__ hfa384x_MaxTransmitLifetime_t;
+} __attribute__((packed)) hfa384x_MaxTransmitLifetime_t;
 
 /*-- Information Record: MaxReceiveLifetime --*/
 typedef struct hfa384x_MaxReceiveLifetime
 {
 	u16	MaxReceiveLifetime;
-} __WLAN_ATTRIB_PACK__ hfa384x_MaxReceiveLifetime_t;
+} __attribute__((packed)) hfa384x_MaxReceiveLifetime_t;
 
 /*-- Information Record: CFPollable --*/
 typedef struct hfa384x_CFPollable
 {
 	u16	CFPollable;
-} __WLAN_ATTRIB_PACK__ hfa384x_CFPollable_t;
+} __attribute__((packed)) hfa384x_CFPollable_t;
 
 /*-- Information Record: AuthenticationAlgorithms --*/
 typedef struct hfa384x_AuthenticationAlgorithms
 {
 	u16	AuthenticationType;
 	u16	TypeEnabled;
-} __WLAN_ATTRIB_PACK__ hfa384x_AuthenticationAlgorithms_t;
+} __attribute__((packed)) hfa384x_AuthenticationAlgorithms_t;
 
 /*-- Information Record: AuthenticationAlgorithms
 (data only --*/
@@ -1442,19 +1442,19 @@ typedef struct hfa384x_AuthenticationAlg
 {
 	u16	AuthenticationType;
 	u16	TypeEnabled;
-} __WLAN_ATTRIB_PACK__ hfa384x_AuthenticationAlgorithms_data_t;
+} __attribute__((packed)) hfa384x_AuthenticationAlgorithms_data_t;
 
 /*-- Information Record: PrivacyOptionImplemented --*/
 typedef struct hfa384x_PrivacyOptionImplemented
 {
 	u16	PrivacyOptionImplemented;
-} __WLAN_ATTRIB_PACK__ hfa384x_PrivacyOptionImplemented_t;
+} __attribute__((packed)) hfa384x_PrivacyOptionImplemented_t;
 
 /*-- Information Record: OwnMACAddress --*/
 typedef struct hfa384x_OwnMACAddress
 {
 	u8	OwnMACAddress[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_OwnMACAddress_t;
+} __attribute__((packed)) hfa384x_OwnMACAddress_t;
 
 /*-- Information Record: PCFInfo --*/
 typedef struct hfa384x_PCFInfo
@@ -1463,7 +1463,7 @@ typedef struct hfa384x_PCFInfo
 	u16	CFPPeriod;
 	u16	CFPMaxDuration;
 	u16	CFPFlags;
-} __WLAN_ATTRIB_PACK__ hfa384x_PCFInfo_t;
+} __attribute__((packed)) hfa384x_PCFInfo_t;
 
 /*-- Information Record: PCFInfo (data portion only) --*/
 typedef struct hfa384x_PCFInfo_data
@@ -1472,7 +1472,7 @@ typedef struct hfa384x_PCFInfo_data
 	u16	CFPPeriod;
 	u16	CFPMaxDuration;
 	u16	CFPFlags;
-} __WLAN_ATTRIB_PACK__ hfa384x_PCFInfo_data_t;
+} __attribute__((packed)) hfa384x_PCFInfo_data_t;
 
 /*--------------------------------------------------------------------
 Information Record Structures: Modem Information Records
@@ -1482,31 +1482,31 @@ Information Record Structures: Modem Inf
 typedef struct hfa384x_PHYType
 {
 	u16	PHYType;
-} __WLAN_ATTRIB_PACK__ hfa384x_PHYType_t;
+} __attribute__((packed)) hfa384x_PHYType_t;
 
 /*-- Information Record: CurrentChannel --*/
 typedef struct hfa384x_CurrentChannel
 {
 	u16	CurrentChannel;
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentChannel_t;
+} __attribute__((packed)) hfa384x_CurrentChannel_t;
 
 /*-- Information Record: CurrentPowerState --*/
 typedef struct hfa384x_CurrentPowerState
 {
 	u16	CurrentPowerState;
-} __WLAN_ATTRIB_PACK__ hfa384x_CurrentPowerState_t;
+} __attribute__((packed)) hfa384x_CurrentPowerState_t;
 
 /*-- Information Record: CCAMode --*/
 typedef struct hfa384x_CCAMode
 {
 	u16	CCAMode;
-} __WLAN_ATTRIB_PACK__ hfa384x_CCAMode_t;
+} __attribute__((packed)) hfa384x_CCAMode_t;
 
 /*-- Information Record: SupportedDataRates --*/
 typedef struct hfa384x_SupportedDataRates
 {
 	u8	SupportedDataRates[10];
-} __WLAN_ATTRIB_PACK__ hfa384x_SupportedDataRates_t;
+} __attribute__((packed)) hfa384x_SupportedDataRates_t;
 
 /*-- Information Record: LFOStatus --*/
 typedef struct hfa384x_LFOStatus
@@ -1514,7 +1514,7 @@ typedef struct hfa384x_LFOStatus
 	u16  TestResults;
 	u16  LFOResult;
 	u16  VRHFOResult;
-} __WLAN_ATTRIB_PACK__ hfa384x_LFOStatus_t;
+} __attribute__((packed)) hfa384x_LFOStatus_t;
 
 #define HFA384x_TESTRESULT_ALLPASSED    BIT0
 #define HFA384x_TESTRESULT_LFO_FAIL     BIT1
@@ -1530,7 +1530,7 @@ typedef struct hfa384x_LEDControl
 	u16  assoc_on;
 	u16  assoc_off;
 	u16  activity;
-} __WLAN_ATTRIB_PACK__ hfa384x_LEDControl_t;
+} __attribute__((packed)) hfa384x_LEDControl_t;
 
 /*--------------------------------------------------------------------
                  FRAME DESCRIPTORS AND FRAME STRUCTURES
@@ -1599,7 +1599,7 @@ typedef struct hfa384x_tx_frame
 	u8	dest_addr[6];
 	u8	src_addr[6];
 	u16	data_length; /* big endian format */
-} __WLAN_ATTRIB_PACK__ hfa384x_tx_frame_t;
+} __attribute__((packed)) hfa384x_tx_frame_t;
 /*--------------------------------------------------------------------
 Communication Frames: Field Masks for Transmit Frames
 --------------------------------------------------------------------*/
@@ -1683,7 +1683,7 @@ typedef struct hfa384x_rx_frame
 	u8	dest_addr[6];
 	u8	src_addr[6];
 	u16	data_length; /* IEEE? (big endian) format */
-} __WLAN_ATTRIB_PACK__ hfa384x_rx_frame_t;
+} __attribute__((packed)) hfa384x_rx_frame_t;
 /*--------------------------------------------------------------------
 Communication Frames: Field Masks for Receive Frames
 --------------------------------------------------------------------*/
@@ -1736,7 +1736,7 @@ typedef struct hfa384x_HandoverAddr
 	u16	framelen;
 	u16	infotype;
 	u8	handover_addr[WLAN_BSSID_LEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_HandoverAddr_t;
+} __attribute__((packed)) hfa384x_HandoverAddr_t;
 
 /*--  Inquiry Frame, Diagnose: Communication Tallies --*/
 typedef struct hfa384x_CommTallies16
@@ -1762,7 +1762,7 @@ typedef struct hfa384x_CommTallies16
 	u16	rxdiscardswepundecr;
 	u16	rxmsginmsgfrag;
 	u16	rxmsginbadmsgfrag;
-} __WLAN_ATTRIB_PACK__ hfa384x_CommTallies16_t;
+} __attribute__((packed)) hfa384x_CommTallies16_t;
 
 typedef struct hfa384x_CommTallies32
 {
@@ -1787,7 +1787,7 @@ typedef struct hfa384x_CommTallies32
 	u32	rxdiscardswepundecr;
 	u32	rxmsginmsgfrag;
 	u32	rxmsginbadmsgfrag;
-} __WLAN_ATTRIB_PACK__ hfa384x_CommTallies32_t;
+} __attribute__((packed)) hfa384x_CommTallies32_t;
 
 /*--  Inquiry Frame, Diagnose: Scan Results & Subfields--*/
 typedef struct hfa384x_ScanResultSub
@@ -1801,7 +1801,7 @@ typedef struct hfa384x_ScanResultSub
 	hfa384x_bytestr32_t	ssid;
 	u8	supprates[10]; /* 802.11 info element */
 	u16	proberesp_rate;
-} __WLAN_ATTRIB_PACK__ hfa384x_ScanResultSub_t;
+} __attribute__((packed)) hfa384x_ScanResultSub_t;
 
 typedef struct hfa384x_ScanResult
 {
@@ -1809,7 +1809,7 @@ typedef struct hfa384x_ScanResult
 	u16	scanreason;
 	hfa384x_ScanResultSub_t
 		result[HFA384x_SCANRESULT_MAX];
-} __WLAN_ATTRIB_PACK__ hfa384x_ScanResult_t;
+} __attribute__((packed)) hfa384x_ScanResult_t;
 
 /*--  Inquiry Frame, Diagnose: ChInfo Results & Subfields--*/
 typedef struct hfa384x_ChInfoResultSub
@@ -1818,7 +1818,7 @@ typedef struct hfa384x_ChInfoResultSub
 	u16	anl;
 	u16	pnl;
 	u16	active;
-} __WLAN_ATTRIB_PACK__ hfa384x_ChInfoResultSub_t;
+} __attribute__((packed)) hfa384x_ChInfoResultSub_t;
 
 #define HFA384x_CHINFORESULT_BSSACTIVE	BIT0
 #define HFA384x_CHINFORESULT_PCFACTIVE	BIT1
@@ -1828,7 +1828,7 @@ typedef struct hfa384x_ChInfoResult
 	u16	scanchannels;
 	hfa384x_ChInfoResultSub_t
 		result[HFA384x_CHINFORESULT_MAX];
-} __WLAN_ATTRIB_PACK__ hfa384x_ChInfoResult_t;
+} __attribute__((packed)) hfa384x_ChInfoResult_t;
 
 /*--  Inquiry Frame, Diagnose: Host Scan Results & Subfields--*/
 typedef struct hfa384x_HScanResultSub
@@ -1843,7 +1843,7 @@ typedef struct hfa384x_HScanResultSub
 	u8	supprates[10]; /* 802.11 info element */
 	u16	proberesp_rate;
 	u16	atim;
-} __WLAN_ATTRIB_PACK__ hfa384x_HScanResultSub_t;
+} __attribute__((packed)) hfa384x_HScanResultSub_t;
 
 typedef struct hfa384x_HScanResult
 {
@@ -1851,7 +1851,7 @@ typedef struct hfa384x_HScanResult
 	u16	rsvd;
 	hfa384x_HScanResultSub_t
 		result[HFA384x_HSCANRESULT_MAX];
-} __WLAN_ATTRIB_PACK__ hfa384x_HScanResult_t;
+} __attribute__((packed)) hfa384x_HScanResult_t;
 
 /*--  Unsolicited Frame, MAC Mgmt: LinkStatus --*/
 
@@ -1866,7 +1866,7 @@ typedef struct hfa384x_HScanResult
 typedef struct hfa384x_LinkStatus
 {
 	u16	linkstatus;
-} __WLAN_ATTRIB_PACK__ hfa384x_LinkStatus_t;
+} __attribute__((packed)) hfa384x_LinkStatus_t;
 
 
 /*--  Unsolicited Frame, MAC Mgmt: AssociationStatus (--*/
@@ -1885,7 +1885,7 @@ typedef struct hfa384x_AssocStatus
 	u8	old_ap_addr[WLAN_ADDR_LEN];
 	u16	reason;
 	u16	reserved;
-} __WLAN_ATTRIB_PACK__ hfa384x_AssocStatus_t;
+} __attribute__((packed)) hfa384x_AssocStatus_t;
 
 /*--  Unsolicited Frame, MAC Mgmt: AuthRequest (AP Only) --*/
 
@@ -1893,7 +1893,7 @@ typedef struct hfa384x_AuthRequest
 {
 	u8	sta_addr[WLAN_ADDR_LEN];
 	u16	algorithm;
-} __WLAN_ATTRIB_PACK__ hfa384x_AuthReq_t;
+} __attribute__((packed)) hfa384x_AuthReq_t;
 
 /*--  Unsolicited Frame, MAC Mgmt: AssocRequest (AP Only) --*/
 
@@ -1902,7 +1902,7 @@ typedef struct hfa384x_AssocRequest
 	u8	sta_addr[WLAN_ADDR_LEN];
 	u16	type;
 	u8   wpa_data[80];
-} __WLAN_ATTRIB_PACK__ hfa384x_AssocReq_t;
+} __attribute__((packed)) hfa384x_AssocReq_t;
 
 
 #define HFA384x_ASSOCREQ_TYPE_ASSOC     0
@@ -1914,20 +1914,20 @@ typedef struct hfa384x_MicFailure
 {
 	u8	sender[WLAN_ADDR_LEN];
 	u8	dest[WLAN_ADDR_LEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_MicFailure_t;
+} __attribute__((packed)) hfa384x_MicFailure_t;
 
 /*--  Unsolicited Frame, MAC Mgmt: PSUserCount (AP Only) --*/
 
 typedef struct hfa384x_PSUserCount
 {
 	u16	usercnt;
-} __WLAN_ATTRIB_PACK__ hfa384x_PSUserCount_t;
+} __attribute__((packed)) hfa384x_PSUserCount_t;
 
 typedef struct hfa384x_KeyIDChanged
 {
 	u8	sta_addr[WLAN_ADDR_LEN];
 	u16	keyid;
-} __WLAN_ATTRIB_PACK__ hfa384x_KeyIDChanged_t;
+} __attribute__((packed)) hfa384x_KeyIDChanged_t;
 
 /*--  Collection of all Inf frames ---------------*/
 typedef union hfa384x_infodata {
@@ -1941,14 +1941,14 @@ typedef union hfa384x_infodata {
 	hfa384x_AuthReq_t	authreq;
 	hfa384x_PSUserCount_t	psusercnt;
 	hfa384x_KeyIDChanged_t  keyidchanged;
-} __WLAN_ATTRIB_PACK__ hfa384x_infodata_t;
+} __attribute__((packed)) hfa384x_infodata_t;
 
 typedef struct hfa384x_InfFrame
 {
 	u16			framelen;
 	u16			infotype;
 	hfa384x_infodata_t	info;
-} __WLAN_ATTRIB_PACK__ hfa384x_InfFrame_t;
+} __attribute__((packed)) hfa384x_InfFrame_t;
 
 /*--------------------------------------------------------------------
 USB Packet structures and constants.
@@ -1984,7 +1984,7 @@ USB Packet structures and constants.
 typedef struct hfa384x_usb_txfrm {
 	hfa384x_tx_frame_t	desc;
 	u8			data[WLAN_DATA_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_txfrm_t;
+} __attribute__((packed)) hfa384x_usb_txfrm_t;
 
 typedef struct hfa384x_usb_cmdreq {
 	u16		type;
@@ -1993,21 +1993,21 @@ typedef struct hfa384x_usb_cmdreq {
 	u16		parm1;
 	u16		parm2;
 	u8		pad[54];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_cmdreq_t;
+} __attribute__((packed)) hfa384x_usb_cmdreq_t;
 
 typedef struct hfa384x_usb_wridreq {
 	u16		type;
 	u16		frmlen;
 	u16		rid;
 	u8		data[HFA384x_RIDDATA_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_wridreq_t;
+} __attribute__((packed)) hfa384x_usb_wridreq_t;
 
 typedef struct hfa384x_usb_rridreq {
 	u16		type;
 	u16		frmlen;
 	u16		rid;
 	u8		pad[58];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_rridreq_t;
+} __attribute__((packed)) hfa384x_usb_rridreq_t;
 
 typedef struct hfa384x_usb_wmemreq {
 	u16		type;
@@ -2015,7 +2015,7 @@ typedef struct hfa384x_usb_wmemreq {
 	u16		offset;
 	u16		page;
 	u8		data[HFA384x_USB_RWMEM_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_wmemreq_t;
+} __attribute__((packed)) hfa384x_usb_wmemreq_t;
 
 typedef struct hfa384x_usb_rmemreq {
 	u16		type;
@@ -2023,7 +2023,7 @@ typedef struct hfa384x_usb_rmemreq {
 	u16		offset;
 	u16		page;
 	u8		pad[56];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_rmemreq_t;
+} __attribute__((packed)) hfa384x_usb_rmemreq_t;
 
 /*------------------------------------*/
 /* Response (bulk IN) packet contents */
@@ -2031,12 +2031,12 @@ typedef struct hfa384x_usb_rmemreq {
 typedef struct hfa384x_usb_rxfrm {
 	hfa384x_rx_frame_t	desc;
 	u8			data[WLAN_DATA_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_rxfrm_t;
+} __attribute__((packed)) hfa384x_usb_rxfrm_t;
 
 typedef struct hfa384x_usb_infofrm {
 	u16			type;
 	hfa384x_InfFrame_t	info;
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_infofrm_t;
+} __attribute__((packed)) hfa384x_usb_infofrm_t;
 
 typedef struct hfa384x_usb_statusresp {
 	u16		type;
@@ -2044,7 +2044,7 @@ typedef struct hfa384x_usb_statusresp {
 	u16		resp0;
 	u16		resp1;
 	u16		resp2;
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_cmdresp_t;
+} __attribute__((packed)) hfa384x_usb_cmdresp_t;
 
 typedef hfa384x_usb_cmdresp_t hfa384x_usb_wridresp_t;
 
@@ -2053,7 +2053,7 @@ typedef struct hfa384x_usb_rridresp {
 	u16		frmlen;
 	u16		rid;
 	u8		data[HFA384x_RIDDATA_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_rridresp_t;
+} __attribute__((packed)) hfa384x_usb_rridresp_t;
 
 typedef hfa384x_usb_cmdresp_t hfa384x_usb_wmemresp_t;
 
@@ -2061,17 +2061,17 @@ typedef struct hfa384x_usb_rmemresp {
 	u16		type;
 	u16		frmlen;
 	u8		data[HFA384x_USB_RWMEM_MAXLEN];
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_rmemresp_t;
+} __attribute__((packed)) hfa384x_usb_rmemresp_t;
 
 typedef struct hfa384x_usb_bufavail {
 	u16		type;
 	u16		frmlen;
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_bufavail_t;
+} __attribute__((packed)) hfa384x_usb_bufavail_t;
 
 typedef struct hfa384x_usb_error {
 	u16		type;
 	u16		errortype;
-} __WLAN_ATTRIB_PACK__ hfa384x_usb_error_t;
+} __attribute__((packed)) hfa384x_usb_error_t;
 
 /*----------------------------------------------------------*/
 /* Unions for packaging all the known packet types together */
@@ -2084,7 +2084,7 @@ typedef union hfa384x_usbout {
 	hfa384x_usb_rridreq_t	rridreq;
 	hfa384x_usb_wmemreq_t	wmemreq;
 	hfa384x_usb_rmemreq_t	rmemreq;
-} __WLAN_ATTRIB_PACK__ hfa384x_usbout_t;
+} __attribute__((packed)) hfa384x_usbout_t;
 
 typedef union hfa384x_usbin {
 	u16			type;
@@ -2099,7 +2099,7 @@ typedef union hfa384x_usbin {
 	hfa384x_usb_bufavail_t	bufavail;
 	hfa384x_usb_error_t	usberror;
 	u8			boguspad[3000];
-} __WLAN_ATTRIB_PACK__ hfa384x_usbin_t;
+} __attribute__((packed)) hfa384x_usbin_t;
 
 /*--------------------------------------------------------------------
 PD record structures.
@@ -2108,17 +2108,17 @@ PD record structures.
 typedef struct hfa384x_pdr_pcb_partnum
 {
 	u8	num[8];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_pcb_partnum_t;
+} __attribute__((packed)) hfa384x_pdr_pcb_partnum_t;
 
 typedef struct hfa384x_pdr_pcb_tracenum
 {
 	u8	num[8];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_pcb_tracenum_t;
+} __attribute__((packed)) hfa384x_pdr_pcb_tracenum_t;
 
 typedef struct hfa384x_pdr_nic_serial
 {
 	u8	num[12];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_nic_serial_t;
+} __attribute__((packed)) hfa384x_pdr_nic_serial_t;
 
 typedef struct hfa384x_pdr_mkk_measurements
 {
@@ -2137,12 +2137,12 @@ typedef struct hfa384x_pdr_mkk_measureme
 	double	rx_spur_f2;
 	double	rx_spur_l1;
 	double	rx_spur_l2;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_mkk_measurements_t;
+} __attribute__((packed)) hfa384x_pdr_mkk_measurements_t;
 
 typedef struct hfa384x_pdr_nic_ramsize
 {
 	u8	size[12]; /* units of KB */
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_nic_ramsize_t;
+} __attribute__((packed)) hfa384x_pdr_nic_ramsize_t;
 
 typedef struct hfa384x_pdr_mfisuprange
 {
@@ -2150,7 +2150,7 @@ typedef struct hfa384x_pdr_mfisuprange
 	u16	variant;
 	u16	bottom;
 	u16	top;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_mfisuprange_t;
+} __attribute__((packed)) hfa384x_pdr_mfisuprange_t;
 
 typedef struct hfa384x_pdr_cfisuprange
 {
@@ -2158,7 +2158,7 @@ typedef struct hfa384x_pdr_cfisuprange
 	u16	variant;
 	u16	bottom;
 	u16	top;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_cfisuprange_t;
+} __attribute__((packed)) hfa384x_pdr_cfisuprange_t;
 
 typedef struct hfa384x_pdr_nicid
 {
@@ -2166,140 +2166,140 @@ typedef struct hfa384x_pdr_nicid
 	u16	variant;
 	u16	major;
 	u16	minor;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_nicid_t;
+} __attribute__((packed)) hfa384x_pdr_nicid_t;
 
 
 typedef struct hfa384x_pdr_refdac_measurements
 {
 	u16	value[0];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_refdac_measurements_t;
+} __attribute__((packed)) hfa384x_pdr_refdac_measurements_t;
 
 typedef struct hfa384x_pdr_vgdac_measurements
 {
 	u16	value[0];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_vgdac_measurements_t;
+} __attribute__((packed)) hfa384x_pdr_vgdac_measurements_t;
 
 typedef struct hfa384x_pdr_level_comp_measurements
 {
 	u16	value[0];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_level_compc_measurements_t;
+} __attribute__((packed)) hfa384x_pdr_level_compc_measurements_t;
 
 typedef struct hfa384x_pdr_mac_address
 {
 	u8	addr[6];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_mac_address_t;
+} __attribute__((packed)) hfa384x_pdr_mac_address_t;
 
 typedef struct hfa384x_pdr_mkk_callname
 {
 	u8	callname[8];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_mkk_callname_t;
+} __attribute__((packed)) hfa384x_pdr_mkk_callname_t;
 
 typedef struct hfa384x_pdr_regdomain
 {
 	u16	numdomains;
 	u16	domain[5];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_regdomain_t;
+} __attribute__((packed)) hfa384x_pdr_regdomain_t;
 
 typedef struct hfa384x_pdr_allowed_channel
 {
 	u16	ch_bitmap;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_allowed_channel_t;
+} __attribute__((packed)) hfa384x_pdr_allowed_channel_t;
 
 typedef struct hfa384x_pdr_default_channel
 {
 	u16	channel;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_default_channel_t;
+} __attribute__((packed)) hfa384x_pdr_default_channel_t;
 
 typedef struct hfa384x_pdr_privacy_option
 {
 	u16	available;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_privacy_option_t;
+} __attribute__((packed)) hfa384x_pdr_privacy_option_t;
 
 typedef struct hfa384x_pdr_temptype
 {
 	u16	type;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_temptype_t;
+} __attribute__((packed)) hfa384x_pdr_temptype_t;
 
 typedef struct hfa384x_pdr_refdac_setup
 {
 	u16	ch_value[14];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_refdac_setup_t;
+} __attribute__((packed)) hfa384x_pdr_refdac_setup_t;
 
 typedef struct hfa384x_pdr_vgdac_setup
 {
 	u16	ch_value[14];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_vgdac_setup_t;
+} __attribute__((packed)) hfa384x_pdr_vgdac_setup_t;
 
 typedef struct hfa384x_pdr_level_comp_setup
 {
 	u16	ch_value[14];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_level_comp_setup_t;
+} __attribute__((packed)) hfa384x_pdr_level_comp_setup_t;
 
 typedef struct hfa384x_pdr_trimdac_setup
 {
 	u16	trimidac;
 	u16	trimqdac;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_trimdac_setup_t;
+} __attribute__((packed)) hfa384x_pdr_trimdac_setup_t;
 
 typedef struct hfa384x_pdr_ifr_setting
 {
 	u16	value[3];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_ifr_setting_t;
+} __attribute__((packed)) hfa384x_pdr_ifr_setting_t;
 
 typedef struct hfa384x_pdr_rfr_setting
 {
 	u16	value[3];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_rfr_setting_t;
+} __attribute__((packed)) hfa384x_pdr_rfr_setting_t;
 
 typedef struct hfa384x_pdr_hfa3861_baseline
 {
 	u16	value[50];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_baseline_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_baseline_t;
 
 typedef struct hfa384x_pdr_hfa3861_shadow
 {
 	u32	value[32];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_shadow_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_shadow_t;
 
 typedef struct hfa384x_pdr_hfa3861_ifrf
 {
 	u32	value[20];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_ifrf_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_ifrf_t;
 
 typedef struct hfa384x_pdr_hfa3861_chcalsp
 {
 	u16	value[14];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_chcalsp_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_chcalsp_t;
 
 typedef struct hfa384x_pdr_hfa3861_chcali
 {
 	u16	value[17];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_chcali_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_chcali_t;
 
 typedef struct hfa384x_pdr_hfa3861_nic_config
 {
 	u16	config_bitmap;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_nic_config_t;
+} __attribute__((packed)) hfa384x_pdr_nic_config_t;
 
 typedef struct hfa384x_pdr_hfo_delay
 {
 	u8   hfo_delay;
-} __WLAN_ATTRIB_PACK__ hfa384x_hfo_delay_t;
+} __attribute__((packed)) hfa384x_hfo_delay_t;
 
 typedef struct hfa384x_pdr_hfa3861_manf_testsp
 {
 	u16	value[30];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_manf_testsp_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_manf_testsp_t;
 
 typedef struct hfa384x_pdr_hfa3861_manf_testi
 {
 	u16	value[30];
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_hfa3861_manf_testi_t;
+} __attribute__((packed)) hfa384x_pdr_hfa3861_manf_testi_t;
 
 typedef struct hfa384x_end_of_pda
 {
 	u16	crc;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdr_end_of_pda_t;
+} __attribute__((packed)) hfa384x_pdr_end_of_pda_t;
 
 typedef struct hfa384x_pdrec
 {
@@ -2342,7 +2342,7 @@ typedef struct hfa384x_pdrec
 	hfa384x_pdr_end_of_pda_t	end_of_pda;
 
 	} data;
-} __WLAN_ATTRIB_PACK__ hfa384x_pdrec_t;
+} __attribute__((packed)) hfa384x_pdrec_t;
 
 
 #ifdef __KERNEL__
--- a/drivers/staging/wlan-ng/p80211conv.h
+++ b/drivers/staging/wlan-ng/p80211conv.h
@@ -148,7 +148,7 @@ typedef struct wlan_ethhdr
 	u8	daddr[WLAN_ETHADDR_LEN];
 	u8	saddr[WLAN_ETHADDR_LEN];
 	u16	type;
-} __WLAN_ATTRIB_PACK__ wlan_ethhdr_t;
+} __attribute__((packed)) wlan_ethhdr_t;
 
 /* local llc header type */
 typedef struct wlan_llc
@@ -156,14 +156,14 @@ typedef struct wlan_llc
 	u8	dsap;
 	u8	ssap;
 	u8	ctl;
-} __WLAN_ATTRIB_PACK__ wlan_llc_t;
+} __attribute__((packed)) wlan_llc_t;
 
 /* local snap header type */
 typedef struct wlan_snap
 {
 	u8	oui[WLAN_IEEE_OUI_LEN];
 	u16	type;
-} __WLAN_ATTRIB_PACK__ wlan_snap_t;
+} __attribute__((packed)) wlan_snap_t;
 
 /* Circular include trick */
 struct wlandevice;
--- a/drivers/staging/wlan-ng/p80211hdr.h
+++ b/drivers/staging/wlan-ng/p80211hdr.h
@@ -229,7 +229,7 @@ typedef struct p80211_hdr_a3
 	u8	a2[WLAN_ADDR_LEN];
 	u8	a3[WLAN_ADDR_LEN];
 	u16	seq;
-} __WLAN_ATTRIB_PACK__ p80211_hdr_a3_t;
+} __attribute__((packed)) p80211_hdr_a3_t;
 
 typedef struct p80211_hdr_a4
 {
@@ -240,13 +240,13 @@ typedef struct p80211_hdr_a4
 	u8	a3[WLAN_ADDR_LEN];
 	u16	seq;
 	u8	a4[WLAN_ADDR_LEN];
-} __WLAN_ATTRIB_PACK__ p80211_hdr_a4_t;
+} __attribute__((packed)) p80211_hdr_a4_t;
 
 typedef union p80211_hdr
 {
 	p80211_hdr_a3_t		a3;
 	p80211_hdr_a4_t		a4;
-} __WLAN_ATTRIB_PACK__ p80211_hdr_t;
+} __attribute__((packed)) p80211_hdr_t;
 
 
 /*================================================================*/
--- a/drivers/staging/wlan-ng/p80211ioctl.h
+++ b/drivers/staging/wlan-ng/p80211ioctl.h
@@ -109,7 +109,7 @@ typedef struct p80211ioctl_req
 	u32	magic;
 	u16	len;
 	u32	result;
-} __WLAN_ATTRIB_PACK__ p80211ioctl_req_t;
+} __attribute__((packed)) p80211ioctl_req_t;
 
 
 /*================================================================*/
--- a/drivers/staging/wlan-ng/p80211metastruct.h
+++ b/drivers/staging/wlan-ng/p80211metastruct.h
@@ -55,7 +55,7 @@ typedef struct p80211msg_dot11req_mibget
 	u8		devname[WLAN_DEVNAMELEN_MAX]	;
 	p80211item_unk392_t	mibattribute	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_dot11req_mibget_t;
+} __attribute__((packed)) p80211msg_dot11req_mibget_t;
 
 typedef struct p80211msg_dot11req_mibset
 {
@@ -64,7 +64,7 @@ typedef struct p80211msg_dot11req_mibset
 	u8		devname[WLAN_DEVNAMELEN_MAX]	;
 	p80211item_unk392_t	mibattribute	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_dot11req_mibset_t;
+} __attribute__((packed)) p80211msg_dot11req_mibset_t;
 
 typedef struct p80211msg_dot11req_scan
 {
@@ -85,7 +85,7 @@ typedef struct p80211msg_dot11req_scan
 	p80211item_uint32_t	resultcode	;
 	p80211item_uint32_t	numbss	;
 	p80211item_uint32_t	append	;
-} __WLAN_ATTRIB_PACK__ p80211msg_dot11req_scan_t;
+} __attribute__((packed)) p80211msg_dot11req_scan_t;
 
 typedef struct p80211msg_dot11req_scan_results
 {
@@ -134,7 +134,7 @@ typedef struct p80211msg_dot11req_scan_r
 	p80211item_uint32_t	supprate6	;
 	p80211item_uint32_t	supprate7	;
 	p80211item_uint32_t	supprate8	;
-} __WLAN_ATTRIB_PACK__ p80211msg_dot11req_scan_results_t;
+} __attribute__((packed)) p80211msg_dot11req_scan_results_t;
 
 typedef struct p80211msg_dot11req_start
 {
@@ -173,7 +173,7 @@ typedef struct p80211msg_dot11req_start
 	p80211item_uint32_t	operationalrate7	;
 	p80211item_uint32_t	operationalrate8	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_dot11req_start_t;
+} __attribute__((packed)) p80211msg_dot11req_start_t;
 
 typedef struct p80211msg_lnxreq_ifstate
 {
@@ -182,7 +182,7 @@ typedef struct p80211msg_lnxreq_ifstate
 	u8		devname[WLAN_DEVNAMELEN_MAX]	;
 	p80211item_uint32_t	ifstate	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_lnxreq_ifstate_t;
+} __attribute__((packed)) p80211msg_lnxreq_ifstate_t;
 
 typedef struct p80211msg_lnxreq_wlansniff
 {
@@ -197,7 +197,7 @@ typedef struct p80211msg_lnxreq_wlansnif
 	p80211item_uint32_t	stripfcs	;
 	p80211item_uint32_t	packet_trunc	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_lnxreq_wlansniff_t;
+} __attribute__((packed)) p80211msg_lnxreq_wlansniff_t;
 
 typedef struct p80211msg_lnxreq_hostwep
 {
@@ -207,7 +207,7 @@ typedef struct p80211msg_lnxreq_hostwep
 	p80211item_uint32_t	resultcode	;
 	p80211item_uint32_t	decrypt	;
 	p80211item_uint32_t	encrypt	;
-} __WLAN_ATTRIB_PACK__ p80211msg_lnxreq_hostwep_t;
+} __attribute__((packed)) p80211msg_lnxreq_hostwep_t;
 
 typedef struct p80211msg_lnxreq_commsquality
 {
@@ -219,7 +219,7 @@ typedef struct p80211msg_lnxreq_commsqua
 	p80211item_uint32_t	link	;
 	p80211item_uint32_t	level	;
 	p80211item_uint32_t	noise	;
-} __WLAN_ATTRIB_PACK__ p80211msg_lnxreq_commsquality_t;
+} __attribute__((packed)) p80211msg_lnxreq_commsquality_t;
 
 typedef struct p80211msg_lnxreq_autojoin
 {
@@ -230,7 +230,7 @@ typedef struct p80211msg_lnxreq_autojoin
 	u8	pad_19D[3]	;
 	p80211item_uint32_t	authtype	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_lnxreq_autojoin_t;
+} __attribute__((packed)) p80211msg_lnxreq_autojoin_t;
 
 typedef struct p80211msg_p2req_readpda
 {
@@ -239,7 +239,7 @@ typedef struct p80211msg_p2req_readpda
 	u8		devname[WLAN_DEVNAMELEN_MAX]	;
 	p80211item_unk1024_t	pda	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_p2req_readpda_t;
+} __attribute__((packed)) p80211msg_p2req_readpda_t;
 
 typedef struct p80211msg_p2req_ramdl_state
 {
@@ -249,7 +249,7 @@ typedef struct p80211msg_p2req_ramdl_sta
 	p80211item_uint32_t	enable	;
 	p80211item_uint32_t	exeaddr	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_p2req_ramdl_state_t;
+} __attribute__((packed)) p80211msg_p2req_ramdl_state_t;
 
 typedef struct p80211msg_p2req_ramdl_write
 {
@@ -260,7 +260,7 @@ typedef struct p80211msg_p2req_ramdl_wri
 	p80211item_uint32_t	len	;
 	p80211item_unk4096_t	data	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_p2req_ramdl_write_t;
+} __attribute__((packed)) p80211msg_p2req_ramdl_write_t;
 
 typedef struct p80211msg_p2req_flashdl_state
 {
@@ -269,7 +269,7 @@ typedef struct p80211msg_p2req_flashdl_s
 	u8		devname[WLAN_DEVNAMELEN_MAX]	;
 	p80211item_uint32_t	enable	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_p2req_flashdl_state_t;
+} __attribute__((packed)) p80211msg_p2req_flashdl_state_t;
 
 typedef struct p80211msg_p2req_flashdl_write
 {
@@ -280,6 +280,6 @@ typedef struct p80211msg_p2req_flashdl_w
 	p80211item_uint32_t	len	;
 	p80211item_unk4096_t	data	;
 	p80211item_uint32_t	resultcode	;
-} __WLAN_ATTRIB_PACK__ p80211msg_p2req_flashdl_write_t;
+} __attribute__((packed)) p80211msg_p2req_flashdl_write_t;
 
 #endif
--- a/drivers/staging/wlan-ng/p80211mgmt.h
+++ b/drivers/staging/wlan-ng/p80211mgmt.h
@@ -247,7 +247,7 @@ typedef struct wlan_ie
 {
 	u8	eid;
 	u8	len;
-} __WLAN_ATTRIB_PACK__ wlan_ie_t;
+} __attribute__((packed)) wlan_ie_t;
 
 /*-- Service Set Identity (SSID)  -----------------*/
 typedef struct wlan_ie_ssid
@@ -255,7 +255,7 @@ typedef struct wlan_ie_ssid
 	u8	eid;
 	u8	len;
 	u8	ssid[1];  /* may be zero, ptrs may overlap */
-} __WLAN_ATTRIB_PACK__ wlan_ie_ssid_t;
+} __attribute__((packed)) wlan_ie_ssid_t;
 
 /*-- Supported Rates  -----------------------------*/
 typedef struct wlan_ie_supp_rates
@@ -263,7 +263,7 @@ typedef struct wlan_ie_supp_rates
 	u8	eid;
 	u8	len;
 	u8	rates[1];  /* had better be at LEAST one! */
-} __WLAN_ATTRIB_PACK__ wlan_ie_supp_rates_t;
+} __attribute__((packed)) wlan_ie_supp_rates_t;
 
 /*-- FH Parameter Set  ----------------------------*/
 typedef struct wlan_ie_fh_parms
@@ -274,7 +274,7 @@ typedef struct wlan_ie_fh_parms
 	u8	hopset;
 	u8	hoppattern;
 	u8	hopindex;
-} __WLAN_ATTRIB_PACK__ wlan_ie_fh_parms_t;
+} __attribute__((packed)) wlan_ie_fh_parms_t;
 
 /*-- DS Parameter Set  ----------------------------*/
 typedef struct wlan_ie_ds_parms
@@ -282,7 +282,7 @@ typedef struct wlan_ie_ds_parms
 	u8	eid;
 	u8	len;
 	u8	curr_ch;
-} __WLAN_ATTRIB_PACK__ wlan_ie_ds_parms_t;
+} __attribute__((packed)) wlan_ie_ds_parms_t;
 
 /*-- CF Parameter Set  ----------------------------*/
 
@@ -294,7 +294,7 @@ typedef struct wlan_ie_cf_parms
 	u8	cfp_period;
 	u16	cfp_maxdur;
 	u16	cfp_durremaining;
-} __WLAN_ATTRIB_PACK__ wlan_ie_cf_parms_t;
+} __attribute__((packed)) wlan_ie_cf_parms_t;
 
 /*-- TIM ------------------------------------------*/
 typedef struct wlan_ie_tim
@@ -305,7 +305,7 @@ typedef struct wlan_ie_tim
 	u8	dtim_period;
 	u8	bitmap_ctl;
 	u8	virt_bm[1];
-} __WLAN_ATTRIB_PACK__ wlan_ie_tim_t;
+} __attribute__((packed)) wlan_ie_tim_t;
 
 /*-- IBSS Parameter Set ---------------------------*/
 typedef struct wlan_ie_ibss_parms
@@ -313,7 +313,7 @@ typedef struct wlan_ie_ibss_parms
 	u8	eid;
 	u8	len;
 	u16	atim_win;
-} __WLAN_ATTRIB_PACK__ wlan_ie_ibss_parms_t;
+} __attribute__((packed)) wlan_ie_ibss_parms_t;
 
 /*-- Challenge Text  ------------------------------*/
 typedef struct wlan_ie_challenge
@@ -321,7 +321,7 @@ typedef struct wlan_ie_challenge
 	u8	eid;
 	u8	len;
 	u8	challenge[1];
-} __WLAN_ATTRIB_PACK__ wlan_ie_challenge_t;
+} __attribute__((packed)) wlan_ie_challenge_t;
 
 /*-------------------------------------------------*/
 /*  Frame Types  */
--- a/drivers/staging/wlan-ng/p80211msg.h
+++ b/drivers/staging/wlan-ng/p80211msg.h
@@ -81,7 +81,7 @@ typedef struct p80211msg
 	u32	msgcode;
 	u32	msglen;
 	u8	devname[WLAN_DEVNAMELEN_MAX];
-} __WLAN_ATTRIB_PACK__ p80211msg_t;
+} __attribute__((packed)) p80211msg_t;
 
 typedef struct p80211msgd
 {
@@ -89,7 +89,7 @@ typedef struct p80211msgd
 	u32	msglen;
 	u8	devname[WLAN_DEVNAMELEN_MAX];
 	u8	args[0];
-} __WLAN_ATTRIB_PACK__ p80211msgd_t;
+} __attribute__((packed)) p80211msgd_t;
 
 /*================================================================*/
 /* Extern Declarations */
--- a/drivers/staging/wlan-ng/p80211types.h
+++ b/drivers/staging/wlan-ng/p80211types.h
@@ -395,48 +395,48 @@ typedef struct p80211enum
 typedef struct p80211pstr
 {
 	u8		len;
-} __WLAN_ATTRIB_PACK__ p80211pstr_t;
+} __attribute__((packed)) p80211pstr_t;
 
 typedef struct p80211pstrd
 {
 	u8		len;
 	u8		data[0];
-} __WLAN_ATTRIB_PACK__ p80211pstrd_t;
+} __attribute__((packed)) p80211pstrd_t;
 
 /* Maximum pascal string */
 typedef struct p80211pstr255
 {
 	u8		len;
 	u8		data[MAXLEN_PSTR255];
-} __WLAN_ATTRIB_PACK__ p80211pstr255_t;
+} __attribute__((packed)) p80211pstr255_t;
 
 /* pascal string for macaddress and bssid */
 typedef struct p80211pstr6
 {
 	u8		len;
 	u8		data[MAXLEN_PSTR6];
-} __WLAN_ATTRIB_PACK__ p80211pstr6_t;
+} __attribute__((packed)) p80211pstr6_t;
 
 /* pascal string for channel list */
 typedef struct p80211pstr14
 {
 	u8		len;
 	u8		data[MAXLEN_PSTR14];
-} __WLAN_ATTRIB_PACK__ p80211pstr14_t;
+} __attribute__((packed)) p80211pstr14_t;
 
 /* pascal string for ssid */
 typedef struct p80211pstr32
 {
 	u8		len;
 	u8		data[MAXLEN_PSTR32];
-} __WLAN_ATTRIB_PACK__ p80211pstr32_t;
+} __attribute__((packed)) p80211pstr32_t;
 
 /* MAC address array */
 typedef struct p80211macarray
 {
 	u32		cnt;
 	u8		data[1][MAXLEN_PSTR6];
-} __WLAN_ATTRIB_PACK__ p80211macarray_t;
+} __attribute__((packed)) p80211macarray_t;
 
 /* prototype template */
 typedef struct p80211item
@@ -444,7 +444,7 @@ typedef struct p80211item
 	u32		did;
 	u16		status;
 	u16		len;
-} __WLAN_ATTRIB_PACK__ p80211item_t;
+} __attribute__((packed)) p80211item_t;
 
 /* prototype template w/ data item */
 typedef struct p80211itemd
@@ -453,7 +453,7 @@ typedef struct p80211itemd
 	u16		status;
 	u16		len;
 	u8		data[0];
-} __WLAN_ATTRIB_PACK__ p80211itemd_t;
+} __attribute__((packed)) p80211itemd_t;
 
 /* message data item for int, BOUNDEDINT, ENUMINT */
 typedef struct p80211item_uint32
@@ -462,7 +462,7 @@ typedef struct p80211item_uint32
 	u16		status;
 	u16		len;
 	u32		data;
-} __WLAN_ATTRIB_PACK__ p80211item_uint32_t;
+} __attribute__((packed)) p80211item_uint32_t;
 
 /* message data item for OCTETSTR, DISPLAYSTR */
 typedef struct p80211item_pstr6
@@ -471,7 +471,7 @@ typedef struct p80211item_pstr6
 	u16		status;
 	u16		len;
 	p80211pstr6_t	data;
-} __WLAN_ATTRIB_PACK__ p80211item_pstr6_t;
+} __attribute__((packed)) p80211item_pstr6_t;
 
 /* message data item for OCTETSTR, DISPLAYSTR */
 typedef struct p80211item_pstr14
@@ -480,7 +480,7 @@ typedef struct p80211item_pstr14
 	u16			status;
 	u16			len;
 	p80211pstr14_t		data;
-} __WLAN_ATTRIB_PACK__ p80211item_pstr14_t;
+} __attribute__((packed)) p80211item_pstr14_t;
 
 /* message data item for OCTETSTR, DISPLAYSTR */
 typedef struct p80211item_pstr32
@@ -489,7 +489,7 @@ typedef struct p80211item_pstr32
 	u16			status;
 	u16			len;
 	p80211pstr32_t		data;
-} __WLAN_ATTRIB_PACK__ p80211item_pstr32_t;
+} __attribute__((packed)) p80211item_pstr32_t;
 
 /* message data item for OCTETSTR, DISPLAYSTR */
 typedef struct p80211item_pstr255
@@ -498,7 +498,7 @@ typedef struct p80211item_pstr255
 	u16			status;
 	u16			len;
 	p80211pstr255_t		data;
-} __WLAN_ATTRIB_PACK__ p80211item_pstr255_t;
+} __attribute__((packed)) p80211item_pstr255_t;
 
 /* message data item for UNK 392, namely mib items */
 typedef struct  p80211item_unk392
@@ -507,7 +507,7 @@ typedef struct  p80211item_unk392
 	u16		status;
 	u16		len;
 	u8		data[MAXLEN_MIBATTRIBUTE];
-} __WLAN_ATTRIB_PACK__ p80211item_unk392_t;
+} __attribute__((packed)) p80211item_unk392_t;
 
 /* message data item for UNK 1025, namely p2 pdas */
 typedef struct  p80211item_unk1024
@@ -516,7 +516,7 @@ typedef struct  p80211item_unk1024
 	u16		status;
 	u16		len;
 	u8		data[1024];
-}  __WLAN_ATTRIB_PACK__ p80211item_unk1024_t;
+}  __attribute__((packed)) p80211item_unk1024_t;
 
 /* message data item for UNK 4096, namely p2 download chunks */
 typedef struct  p80211item_unk4096
@@ -525,7 +525,7 @@ typedef struct  p80211item_unk4096
 	u16		status;
 	u16		len;
 	u8		data[4096];
-}  __WLAN_ATTRIB_PACK__ p80211item_unk4096_t;
+}  __attribute__((packed)) p80211item_unk4096_t;
 
 struct catlistitem;
 
--- a/drivers/staging/wlan-ng/wlan_compat.h
+++ b/drivers/staging/wlan-ng/wlan_compat.h
@@ -86,11 +86,6 @@
 #define BIT31	0x80000000
 
 /*=============================================================*/
-/*------ Compiler Portability Macros --------------------------*/
-/*=============================================================*/
-#define __WLAN_ATTRIB_PACK__		__attribute__ ((packed))
-
-/*=============================================================*/
 /*------ OS Portability Macros --------------------------------*/
 /*=============================================================*/
 
